### PR TITLE
Update process

### PR DESCRIPTION
**Pull request by piotrmiskiewicz**
**Description**

Changes proposed in this pull request:
- update process (allows user to modify the instance)
- e2e unit tests which covers REST API
- update OIDC parameters

**Related issue(s)**

### CODE REVIEW

File: components/kyma-environment-broker/cmd/broker/deprovisioning_suite_test.go [hunk 1]
```diff
@@ -78,15 +77,13 @@ func NewDeprovisioningSuite(t *testing.T) *DeprovisioningSuite {
...
 
-	namespaceClientResourceGroupDoesNotExist := azuretesting.NewFakeNamespaceClientResourceGroupDoesNotExist()
-	hyperscalerProvider := azuretesting.NewFakeHyperscalerProvider(namespaceClientResourceGroupDoesNotExist)
 	accountProvider := fixAccountProvider()
 
 	deprovisionManager := deprovisioning.NewManager(db.Operations(), eventBroker, logs.WithField("deprovisioning", "manager"))
 
 	deprovisioningQueue := NewDeprovisioningProcessingQueue(ctx, workersAmount, deprovisionManager, cfg, db, eventBroker,
 		provisionerClient, avsDel, internalEvalAssistant, externalEvalAssistant, smcf,
-		bundleBuilder, edpClient, hyperscalerProvider, accountProvider, logs,
```
Comment (piotrmiskiewicz) [inline components/kyma-environment-broker/cmd/broker/deprovisioning_suite_test.go:89]: hyperscalerProvider was not use in the NewDeprovisioningProcessingQueue

File: components/kyma-environment-broker/internal/broker/instance_update.go [hunk 2]
```diff
@@ -47,15 +55,17 @@ func NewUpdate(cfg Config,
...
 	}
 }
 
 // Update modifies an existing service instance
 //  PATCH /v2/service_instances/{instance_id}
 func (b *UpdateEndpoint) Update(_ context.Context, instanceID string, details domain.UpdateDetails, asyncAllowed bool) (domain.UpdateServiceSpec, error) {
 	logger := b.log.WithField("instanceID", instanceID)
-	logger.Infof("Update instanceID: %s", instanceID)
-	logger.Infof("Update asyncAllowed: %v", asyncAllowed)
+	logger.Infof("Updateing instanceID: %s", instanceID)
```
Comment (ksputo) [inline components/kyma-environment-broker/internal/broker/instance_update.go:66]:
```suggestion
	logger.Infof("Updating instanceID: %s", instanceID)
```

File: components/kyma-environment-broker/internal/broker/instance_update.go [hunk 3]
```diff
@@ -47,15 +55,17 @@ func NewUpdate(cfg Config,
...
+	logger.Infof("Updateing asyncAllowed: %v", asyncAllowed)
```
Comment (ksputo) [inline components/kyma-environment-broker/internal/broker/instance_update.go:67]:
```suggestion
	logger.Infof("Updating asyncAllowed: %v", asyncAllowed)
```

File: components/kyma-environment-broker/internal/broker/instance_update_test.go [hunk 4]
```diff
@@ -203,9 +209,9 @@ func TestUpdateEndpoint_UpdateNonExistingInstance(t *testing.T) {
 	}, true)
 
 	// then
-	assert.IsType(t, err, &apiresponses.FailureResponse{}, "Update returned error of unexpected type")
+	assert.IsType(t, err, &apiresponses.FailureResponse{}, "Updateing returned error of unexpected type")
 	apierr := err.(*apiresponses.FailureResponse)
-	assert.Equal(t, apierr.ValidatedStatusCode(nil), http.StatusNotFound, "Update status code not matching")
+	assert.Equal(t, apierr.ValidatedStatusCode(nil), http.StatusNotFound, "Updateing status code not matching")
```
Comment (ksputo) [inline components/kyma-environment-broker/internal/broker/instance_update_test.go:214]:
```suggestion
	assert.Equal(t, apierr.ValidatedStatusCode(nil), http.StatusNotFound, "Updating status code not matching")
```

File: components/kyma-environment-broker/internal/broker/instance_update_test.go [hunk 5]
```diff
@@ -203,9 +209,9 @@ func TestUpdateEndpoint_UpdateNonExistingInstance(t *testing.T) {
 	}, true)
 
 	// then
-	assert.IsType(t, err, &apiresponses.FailureResponse{}, "Update returned error of unexpected type")
+	assert.IsType(t, err, &apiresponses.FailureResponse{}, "Updateing returned error of unexpected type")
```
Comment (ksputo) [inline components/kyma-environment-broker/internal/broker/instance_update_test.go:212]:
```suggestion
	assert.IsType(t, err, &apiresponses.FailureResponse{}, "Updating returned error of unexpected type")
```

File: components/kyma-environment-broker/cmd/broker/http_suite_test.go [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+
+	"code.cloudfoundry.org/lager"
+	"github.com/gorilla/mux"
+	"github.com/kyma-project/control-plane/components/kyma-environment-broker/internal/broker"
+	"github.com/kyma-project/control-plane/components/kyma-environment-broker/internal/process"
+	"github.com/kyma-project/control-plane/components/kyma-environment-broker/internal/storage"
+	"github.com/sirupsen/logrus"
+)
+
+// HttpSuite allows to
```
[the remark this answers is not in the document]
Comment (piotrmiskiewicz) [reply components/kyma-environment-broker/cmd/broker/http_suite_test.go:15]: HttpSuite is not used, removed

**Review by ksputo — APPROVED** [review]
lgtm

**Pull request merged**